src/wrapper_generators.jl

```diff
@@ -24,15 +43,15 @@ macro generate_init_header(dependencies...)
     if !isempty(dependencies)
         for dep in dependencies
             push!(deps_path_add, quote
-                append!(PATH_list, $(dep).PATH_list)
-                append!(LIBPATH_list, $(dep).LIBPATH_list)
+                isdefined($(dep), :PATH_list) && append!(PATH_list, $(dep).PATH_list)
+                isdefined($(dep), :LIBPATH_list) && append!(LIBPATH_list, $(dep).LIBPATH_list)
```

Comment:
How can `dep` not be defined?

Reply:
I added a test package in https://github.com/JuliaPackaging/Yggdrasil/pull/4073/files#diff-5e09fe145f6c1c427ef805843582cb41894d5f16009377fa7d69b118a90c043bR72 that defined the Platform called MPIABI.

Here I then inlined the implementation to make it easier to test.

Reply:
Ah, I see. You've added `MPIABI` as a dependency of this JLL package, but it's not a JLL package.

We.... have kind of assumed so far that JLL packages only depend on other JLL packages.  That made it very easy to do things like avoid package loading loops.

I'm trying to find where `MPIABI` is defined, but I can't find it in General?

Reply:
https://github.com/JuliaParallel/MPI.jl/pull/529

See the registry at: https://GitHub.com/vchuravy/Experimental